### PR TITLE
Revamp docs

### PR DESCRIPTION
**Pull request by nhooyr**
- [x] Revamp README.md
- [x] Finish ./doc/guide.md
- [x] Clarify npm install deps

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,47 +1,93 @@
 # code-server
```
Comment (nhooyr) [inline README.md:1]: @code-asher only review README.md for now pls

**Comment by nhooyr** [comment]
should split test into fmt, lint and test.

File: README.md [hunk 2]
```diff
@@ -1,47 +1,94 @@
...
+```
 
-## Getting Started
+### npm
 
-### Requirements
+```bash
+npm install -g code-server
+code-server
+```
```
Comment (code-asher) [inline README.md:41]: To match the others should we put the `Now visit` line here?

Comment (nhooyr) [reply README.md:41]: Opted against so that user's would read the output but I'll add for visual consistency.

Comment (code-asher) [reply README.md:41]: That makes sense.

**Comment by nhooyr** [comment]
So idk wtf is going on with both npm/yarn, but you can't publish any `node_modules` folders in a module. So now we rename them all to `node_modules.bundled` and then on post install rename them back.

Also have to unset some environment variables for `npm rebuild` to work under `npm -g`. Now all works well :)

**Comment by nhooyr** [comment]
Punting windows release for now, this stuff took a lot longer than expected. Going to just finish this guide and then we should release.

**Comment by nhooyr** [comment]
- [ ] If npm rebuild fails, should print a link to our npm docs

**Comment by nhooyr** [comment]
- [ ] Discuss 8443 as default port or enabling TLS by default and using 443 potentially 🤔

**Comment by nhooyr** [comment]
- [ ] Move old extension directory on macOS

**Comment by nhooyr** [comment]
I do want to avoid an automagic updating related to their data. Maybe best we just leave it but mention it in the release notes? @code-asher

**Comment by nhooyr** [comment]
Updated the FAQ and added the getting started guide! I think we're reading for v3.3.0.

Please review @cdr/coconuts

**Comment by nhooyr** [comment]
Tomorrow I'll test out all the instructions and make sure everything works as expected.

**Comment by nhooyr** [comment]
@code-asher We should get to together tomorrow and draft the release notes as well, indicating all minor breaking changes

**Comment by nhooyr** [comment]
@code-asher Please review the draft release notes, I added all the minor breaking changes I could think of

**Comment by nhooyr** [comment]
@cdr/coconuts .yml or .yaml???

**Comment by nhooyr** [comment]
- [ ] Add config entry for the code server config file.

**Comment by nhooyr** [comment]
- [ ] FAQ entry on data directory

**Comment by nhooyr** [comment]
- [ ] Add back code-server script into root for now

**Pull request merged**